Modify S1862: LaYC format (#2311)
SonarSource/rspec: rules/S1862/csharp/rule.adoc

== Why is this an issue?

> A chain of https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/selection-statements#the-if-statement[if/else if] statements is evaluated from top to bottom. At most, only one branch will be executed: the first one with a condition that evaluates to `true`. 
> Therefore, duplicating a condition automatically leads to dead code inside the duplicated condition block. Usually, this is due to a copy/paste error. 

Reviewer: ```suggestion
A chain of https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/selection-statements#the-if-statement[if/else if] statements are evaluated from top to bottom. At most, only one branch will be executed: the first whose statement with the condition that evaluates to `true`. 
Therefore, duplicating a condition leads to unreachable code inside the duplicated condition block. Usually, this is due to a copy/paste error. 
```

Author: I don't agree with grammarly about the `is` -> `are`. I think the `(chain of statements)` as a unit is one thing, so it should be `is evaluated` and not `are evaluated`.

Reviewer: True - I aggree.

> At best, it's simply dead code and at worst, it's a bug that could lead to unexpected behavior. 

Reviewer: ```suggestion
The result of such duplication can lead to unreachable code or even to unexpected behavior. 
```

{
  CloseWindow();
}
else if (param == 1) // Noncompliant: condition has already been checked
{

>   MoveWindowToTheBackground(); // dead code

Reviewer: ```suggestion
  MoveWindowToTheBackground(); // unreachable code
```

Author: I think at some point we were told to use semi-realistic examples (e.g. not use `foo` and `bar` with no meaning that much).

This example seems straightforward to me, and I cannot think of a simpler one that makes sense. If you have any idea, please let me know :)

Reviewer: I find `MoveWindowToTheBackground` for example a bit out of place. But I'm not too strong on that - it still passes the message and is simple enought.
